src/applications/personalization/profile/tests/e2e/contact-information/confirm-email-address.cypress.spec.js

```diff
@@ -0,0 +1,243 @@
+    statusCode: 200,
+    body: mockUser,
+  }).as('finalUserRequest');
+
+  cy.visit(PROFILE_PATHS.CONTACT_INFORMATION);
+  cy.wait('@mockUser');
+};
+
+const clickConfirmEmail = () => {
+  cy.get(
+    'va-alert[data-testid="profile-alert--confirm-contact-email"], va-alert[data-testid="mhv-alert--confirm-error',
+  )
+    .find('va-button-pair')
+    .shadow()
+    .find('va-button')
+    .first()
+    .shadow()
+    .find('button')
+    .click();
+};
+
+const clickEditEmail = () => {
+  cy.get(
+    'va-alert[data-testid="profile-alert--confirm-contact-email"], va-alert[data-testid="mhv-alert--confirm-error',
+  )
+    .find('va-button-pair')
+    .shadow()
+    .within(() => {
+      cy.get('va-button[secondary]')
+        .shadow()
+        .find('button')
+        .click();
+    });
+};
+
+const buildUpdateEmailResponse = (isSuccess = true) => ({
+  statusCode: isSuccess ? 200 : 400,
+  body: {
+    data: {
+      id: '',
+      type: 'async_transaction_va_profile_email_address_transactions',
+      attributes: {
+        transactionId: 'email_address_tx_id',
+        transactionStatus: isSuccess ? 'RECEIVED' : 'COMPLETED_FAILURE',
+        type: 'AsyncTransaction::VAProfile::EmailAddressTransaction',
+        metadata: [],
+      },
+    },
+  },
+});
+
+describe('MHV Email Confirmation Alert - Confirm Email', () => {
+  beforeEach(() => {
+    setup();
+    cy.clearCookie(MHV_EMAIL_CONFIRMATION_DISMISSED_COOKIE);
+  });
+
+  it('should not show the error alert when the feature flag is disabled', () => {
+    mockFeatureToggles(() => ({
+      data: {
+        type: 'feature_toggles',
+        features: [
+          {
+            name: 'mhvEmailConfirmation',
+            value: false,
+          },
+        ],
+      },
+    }));
+
+    cy.findByTestId('profile-alert--confirm-contact-email').should('not.exist');
+
+    cy.injectAxeThenAxeCheck();
+  });
+
+  it('should show the error alert when confirm fails and handle a successful retry attempt', () => {
+    let hasFailed = false;
+    cy.intercept('PUT', '/v0/profile/email_addresses', req => {
+      if (!hasFailed) {
+        hasFailed = true;
+        req.reply(buildUpdateEmailResponse(false));
+      } else {
+        req.reply(buildUpdateEmailResponse());
+      }
+    }).as('updateEmail');
+
+    clickConfirmEmail();
+    cy.wait('@updateEmail');
+
+    // Confirm error alert appears in edit view
+    cy.get('[data-testid="mhv-alert--confirm-contact-email"]').should(
+      'not.exist',
+    );
+    cy.findByTestId('mhv-alert--confirm-error').should('be.visible');
+    cy.findByTestId('mhv-alert--confirm-error').should('be.focused');
+
+    // Verify the MHV email confirmation alert cookie was NOT set
+    cy.getCookie(MHV_EMAIL_CONFIRMATION_DISMISSED_COOKIE).should('be.null');
+
+    clickConfirmEmail();
+    cy.wait('@updateEmail');
+
+    // Confirm success alert shows after successful retry
+    cy.get('[data-testid="mhv-alert--confirm-contact-email"]').should(
```

Comment:
This needs to be a find as the page may not be rendered yet even with the cy.wait above.